src/metatrain/experimental/pet/model.py

```diff
@@ -61,6 +61,12 @@ def __init__(self, model_hypers: Dict, dataset_info: DatasetInfo) -> None:
         self.is_lora_applied = False
         self.checkpoint_path: Optional[str] = None
 
+        # last-layer feature size (for LLPR module)
+        self.last_layer_feature_size = (
+            self.hypers["N_GNN_LAYERS"] * self.hypers["HEAD_N_NEURONS"] * 2
```

Comment:
Is this correct? In the pure pet code, there code for calculating LLFs looks like this

```python
last_layer_features = []
if "central_token" in result.keys():
    predictor_output = central_tokens_predictor(
                    result["central_token"], central_species
                )
    last_layer_features.append(predictor_output["features"])
if self.USE_BOND_ENERGIES:
    predictor_output = messages_bonds_predictor(
                    output_messages, mask, nums, central_species, multipliers
                )
    last_layer_features.append(predictor_output["features"])

last_layer_features = torch.concatenate(last_layer_features, dim=1)
last_layer_features = torch_geometric.nn.global_mean_pool(last_layer_features, batch=batch_dict["batch"])
```

So it's not necessarily multiplied by 2, only if `self.USE_BOND_ENERGIES` is True

Reply:
Thanks a lot, you're right